### PR TITLE
use AD to compute shape derivatives

### PR DESCRIPTION
**Pull request by KristofferC**

### CODE REVIEW

File: src/JuAFEM.jl [hunk 1]
```diff
@@ -3,6 +3,7 @@ __precompile__()
 module JuAFEM
 using FastGaussQuadrature
 using Compat
+using ForwardDiff
```
Comment (fredrikekre) [inline src/JuAFEM.jl]: Why can we not use Tensors only here?

File: src/interpolations.jl [hunk 2]
```diff
@@ -115,7 +128,7 @@ getnbasefunctions(::Lagrange{1, RefCube, 1}) = 2
 getnfacenodes(::Lagrange{1, RefCube, 1}) = 1
 getfacelist(::Type{Lagrange{1, RefCube, 1}}) = ((1,),(2,))
 
-function value!(ip::Lagrange{1, RefCube, 1}, N::AbstractVector, ξ::Vec{1})
+function value!(ip::Lagrange{1, RefCube, 1}, N::AbstractVector, ξ::AbstractVector)
```
Comment (fredrikekre) [inline src/interpolations.jl]: is the quadpoints not Vec anymore?

Comment (KristofferC) [inline src/interpolations.jl]: They are but not when ForwardDiff calls this function

Comment (fredrikekre) [inline src/interpolations.jl]: Right. If we called `value!` for each independent shape function we could have used AD from `Tensors`, right?

**Pull request merged**